renpy/revertable.py

```diff
@@ -411,10 +411,54 @@ def _rollback(self, compressed):
         set.update(self, compressed)
 
 
-class RevertableObject(object):
+class RevertableObjectMeta(type):
+    def __new__(cls, name, bases, dct):
+        rv = super(RevertableObjectMeta, cls).__new__(cls, name, bases, dct)
+
+        # Collect all slots from type MRO
+        slots = set()
+        for t in rv.__mro__:
+            # Stop on object as it has no slots but it is special
+            if t is object:
+                break
+
+            if "__slots__" in t.__dict__:
+                slots.update(t.__slots__)
+            else:
+                slots.add("__dict__")
```

Comment:
Proposed fix to avoid capturing slots defined in subclasses whose ancestors have already dropped `__slots__`.
```suggestion
        for t in reversed(rv.__mro__):
            # Skip object as it has no slots but it is special
            if t is object:
                continue

            if '__slots__' not in t.__dict__:
                slots.add('__dict__')
                break

            slots.update(t.__slots__)
```